dwarf-op.lua

```diff
@@ -1008,32 +1013,46 @@ end
 local includeProtectedDwfs = false
 function CheckWorker(dwf, option)
     if CanWork(dwf) then
+        local name = dfhack.TranslateName(dfhack.units.getVisibleName(dwf))
+        local nickname = dwf.status.current_soul.name.nickname
         --check option data type (string/table)
-            --string:
+            --normal
                 --check if option specifies a pattern which matches the name of this dwf
                 --check if we want highlighted dwf & whether that is this dwf
                 --check if option starts with 'p'
                 --check all the possible options
-            --table:
+            --list:
                 --check if option[1] starts with 'p'
                 --check all possible options
+        local list
         if type(option) == 'string' then
-            local name = dfhack.TranslateName(dfhack.units.getVisibleName(dwf))
-            local nickname = dwf.status.current_soul.name.nickname
+            list = argparse.stringList(option)
+        else
+            error("The select option entered is not a string.")
+        end
+        local N = #list
+
+        -- parse the selection arguments
+        if N == 1 then
+            -- name
             if string.match(name,option) or string.match(nickname,option) then
                 return true
```

Comment:
I don't understand why this is only being run if `#list == 1` -- what if the user specifies something like your example of `bob,james,urist`?

Reply:
although really it might keep things simple if the user could pass only one name. that would make it easier to distinguish when the 'p' prefix should be given its special meaning. Unless the name to be matched is "pall" or something..

Reply:
`N==1` is the equivalent of what `type(option) == 'string'` was.

> what if the user specifies something like your example..

See line 1088.

And if the user enters only one name, and it's something like pall or has any keyword inside of it.. this would have always broken in that edge case. They should avoid naming their dwarves `drunk` for example. What it would do is, it would select all dwarves with the drunk profession, and the ones named drunk. So it will break in that way if they use keywords to name dwarves. I've never seen a dwarf with `all` as a substring of their name, but if that happens naturally.. and they are protected.. it'll select them anyway regardless of whether they have the `p` prefix.

They should also avoid using numbers in the names, or jobs (the two primary list-able options). With numbers in their names, and trying to select by waves, the same thing will happen. It'll select the waves, and dwarves with it in their name. It'll always select what they want.. but it may select extra if they choose to name their dwarves in non-recommended ways. This is one of the reasons `--show` exists so the user can verify the selected dwarves.

You do bring up a valid point about the p prefix though, since it always checks the first option for a `p` prefix.. providing a list `paul,brent,davy` would enable the protected selection flag.. However since choosing by name always ignores protected status, it is a non-issue. Anything else and they are in a non-recommended edge case where they're integrating things into names they probably shouldn't be.

Reply:
> `N==1` is the equivalent of what `type(option) == 'string'` was.

That would be true if this line appeared above `list = argparse.stringList(option)`, but where the test is currently located, if `option` is `paul,brent,davy` then `#list` will be 3, not 1.

Reply:
Gotcha, though it seems like we are unnecessarily restricting what players can name their dwarves. This restriction could have been avoided by starting the name list with a keyword like "name".

Reply:
I can't remember if there was some way of thinking that led me to want this particular behaviour, but I can think on it tomorrow before I test my spectate unpause code.

I'll probably add a keyword, or figure out what I was thinking and agree with it. In the latter I'll give you the opportunity to show me the err of my thoughts :)

Reply:
I think maybe I wanted to embed the names with other selection options.. but I had restricted it to one name, so that kinda rules that out and maybe even implies had actually considered this problem and decided to avoid it.

I'll add the keyword. Though it might be nice to be able to select specific dwarves along side particular jobs or waves, probably better to let the eccentric guy name their dwarves: {{dwarf1,dwarf2,...,dwarfN}|{Miner1,Farmer3,Admin93}} etc.